### PR TITLE
Remove usage of BRAVE_CHROMIUM_BUILD and brave_chromium_build

### PR DESCRIPTION
**Pull request by mariospr**
It will assumed from now on that all of Brave's own code plus patches
to upstream Chromium will always be built and run as part of a Brave
build, so we can drop these guards and reduce the patching surface.

Note that github.com/brave/brave-browser/wiki/Patching-Chromium should
also be updated to reflect the change implemented in this patch.

Resolves https://github.com/brave/brave-browser/issues/12740

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

N/A

### CODE REVIEW

File: patches/components-update_client-updater_state_win.cc.patch
Comment (iefremov) [inline patches/components-update_client-updater_state_win.cc.patch:25]: for future work: all of these potentitally can be redefined via chromium_src?

File: patches/chrome-common-channel_info_win.cc.patch
Comment (iefremov) [inline patches/chrome-common-channel_info_win.cc.patch:8]: candidate for chromium_src?

File: patches/chrome-browser-search_engines-template_url_service_factory.cc.patch
Comment (iefremov) [inline patches/chrome-browser-search_engines-template_url_service_factory.cc.patch:9]: chromium_src

File: patches/chrome-browser-net-net_error_tab_helper.cc.patch
Comment (iefremov) [inline patches/chrome-browser-net-net_error_tab_helper.cc.patch:9]: maybe we could develop a way to avoid creation of certain tabhelpers, because it seems the whole thing is not needed for us

**Review by iefremov — APPROVED** [review]
Looks good to me, and it seems that there are few patches that can be converted to chormium_src files, especially the ones with constants (I've highlighted a few, but obviously there are more)

File: patches/components-update_client-updater_state_win.cc.patch
Comment (mariospr) [reply patches/components-update_client-updater_state_win.cc.patch:25]: Yes, I had the same thought exactly, but for this patch I think it's probably better to keep it like this so that it's more clear what the change is about?

File: patches/chrome-common-channel_info_win.cc.patch
Comment (mariospr) [reply patches/chrome-common-channel_info_win.cc.patch:8]: Indeed, will work on this (and other cases) in follow-up PRs if that's ok

File: patches/chrome-browser-search_engines-template_url_service_factory.cc.patch
Comment (mariospr) [reply patches/chrome-browser-search_engines-template_url_service_factory.cc.patch:9]: Ditto (re: doing this in a follow-up PR)

File: patches/chrome-browser-net-net_error_tab_helper.cc.patch
Comment (mariospr) [reply patches/chrome-browser-net-net_error_tab_helper.cc.patch:9]: I think I lack context to properly understand what you mean here. Could you please elaborate a bit more?

Comment (iefremov) [reply patches/chrome-browser-net-net_error_tab_helper.cc.patch:9]: It appears to me that the whole point of this class is sending some sort of "probes" which we disable in this method. So instead of patching we could maybe avoid instantiating the helper (it's unclear to me how exactly, just a raw idea). And probably there are more unneeded helpers (or keyed services) like this.

**Comment by mariospr** [comment]
> Looks good to me, and it seems that there are few patches that can be converted to chormium_src files, especially the ones with constants (I've highlighted a few, but obviously there are more)

@iefremov Thanks for the review! I understand you're ok with doing the chromium_src suggested changes in a separate PR, so I'll work on that next, but let me know whether there's something else you want me to change in this particular one. Thanks!

File: patches/chrome-browser-net-net_error_tab_helper.cc.patch
Comment (mariospr) [reply patches/chrome-browser-net-net_error_tab_helper.cc.patch:9]: I see what you mean now, thanks for the clarification.

**Comment by iefremov** [comment]
@mariospr I'm fine with this one, but it is patch masters who decide :)

File: patches/chrome-browser-net-net_error_tab_helper.cc.patch
Comment (mariospr) [reply patches/chrome-browser-net-net_error_tab_helper.cc.patch:9]: Handled in https://github.com/brave/brave-browser/issues/14219

File: patches/chrome-browser-search_engines-template_url_service_factory.cc.patch
Comment (mariospr) [reply patches/chrome-browser-search_engines-template_url_service_factory.cc.patch:9]: Tracked in https://github.com/brave/brave-core/pull/7980

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc:6]: Unrelated to this PR, but we should probably include all headers from the original file here.

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h
Comment (mkarolin) [inline chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h:9]: Same, unrelated to this PR, but we should probably include all headers from the original file here.

File: patches/chrome-BUILD.gn.patch
Comment (mkarolin) [inline patches/chrome-BUILD.gn.patch:10]: Can collapse into a single line.

File: patches/chrome-BUILD.gn.patch
Comment (mkarolin) [inline patches/chrome-BUILD.gn.patch:79]: Can collapse into a single line.

File: patches/chrome-browser-profiles-profile_io_data.cc.patch
Comment (mkarolin) [inline patches/chrome-browser-profiles-profile_io_data.cc.patch:9]: Unrelated to this PR, this can probably be done via a `chromium_src` override.

File: patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch
Comment (mkarolin) [inline patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch:12]: Wonder if we can simplify this by using something like `true ? FILE_PATH_LITERAL("brave.exe") :`

**Review by mkarolin — APPROVED** [review]
++
Need to update https://github.com/brave/brave-browser/wiki/Patching-Chromium re: using BRAVE_CHROMIUM_BUILD.

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc
Comment (mariospr) [reply chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc:6]: I'll prepare a separate PR for these changes. Thanks

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h
Comment (mariospr) [reply chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h:9]: Ditto (i.e. will do on a separate PR)

File: patches/chrome-BUILD.gn.patch
Comment (mariospr) [reply patches/chrome-BUILD.gn.patch:10]: Hmmm... how do we do that? Note that one it's an addition and the other one is a removal :)

File: patches/chrome-browser-profiles-profile_io_data.cc.patch
Comment (mariospr) [reply patches/chrome-browser-profiles-profile_io_data.cc.patch:9]: I actually looked into doing that yesterday, but it was not clear to me how we would do it besides creating a define in the chromium_src override and then patching this file to include that define. It would be 1 line vs 3 lines in the patch, though, so perhaps still worth it.

@mkarolin Could you clarify what you mean?

File: patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch
Comment (mariospr) [reply patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch:12]: That would certainly work too. I'll change it if that's preferred.

File: patches/chrome-BUILD.gn.patch
Comment (mariospr) [reply patches/chrome-BUILD.gn.patch:10]: Ah, ok... you mean putting the two statements in the same line, not merging the two lists. Ok

File: patches/chrome-BUILD.gn.patch
Comment (mariospr) [reply patches/chrome-BUILD.gn.patch:79]: Ok

File: patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch
Comment (mariospr) [reply patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch:12]: Actually, I think I'll leave it as I had it for now if you don't mind, since doing what you suggest would go against some of the rules we have in the ["Patching Chromium" wiki page](https://github.com/brave/brave-browser/wiki/Patching-Chromium):

  _"If possible write the patch to add a new line vs appending/prepending to an existing line."_

I'm aware though that your suggestion means less lines added in the patch, but my hesitation comes from the fact that, if we're going to modify the existing line, we could very well then simply replace the string doing something like this:
```diff
- FILE_PATH_LITERAL("chrome.exe");
+ FILE_PATH_LITERAL("brave.exe");
```
WDYT?

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc
Comment (mariospr) [reply chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.cc:15]: On a second thought, I think the change is small enough we can include it here (I thought there would be many other cases defining `GOOGLE_CHROME_BUILD` in other chromium_src overrides, but turns out it's only these two files). Let me know if you prefer it on a separate PR, though, but otherwise I'm merging that into this PR.

File: chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h
Comment (mariospr) [reply chromium_src/chrome/browser/ui/webui/settings/metrics_reporting_handler.h:13]: As clarified above, the change is small enough (and only affects these files) that I'm including it in this PR in the end.

File: patches/chrome-browser-profiles-profile_io_data.cc.patch
Comment (mariospr) [reply patches/chrome-browser-profiles-profile_io_data.cc.patch:9]: I think I better understood what you meant and got a PR already proposing a change, see https://github.com/brave/brave-core/pull/7995

File: patches/chrome-common-channel_info_win.cc.patch
Comment (mariospr) [reply patches/chrome-common-channel_info_win.cc.patch:8]: Tracked in https://github.com/brave/brave-core/pull/7996

File: patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch
Comment (mariospr) [reply patches/chrome-chrome_proxy-chrome_proxy_main_win.cc.patch:12]: @mkarolin It doesn't seem this is blocking the merge, so I'm going to merge this PR today if it's ok to prevent it from getting conflicts (had to resolve some today already) if it's ok.

**Comment by mariospr** [comment]
The error in the post-init bot (re: npm run audit_deps) is unrelated to this change and this PR can be safely merged as discussed with @mihaiplesa in the morning.

**Review by zenparsing — APPROVED** [review]

**Comment by mariospr** [comment]
Merged and updated the wiki on https://github.com/brave/brave-browser/wiki/Patching-Chromium  (i.e. removed paragraph referencing `BRAVE_CHROMIUM_BUILD`)

File: patches/components-update_client-updater_state_win.cc.patch
Comment (mariospr) [reply patches/components-update_client-updater_state_win.cc.patch:25]: @iefremov I've been looking into this today in the context of other similar situations where I could actually remove the definitions from the .patch file and put them into an override (i.e. https://github.com/brave/brave-core/pull/8063, https://github.com/brave/brave-core/pull/8061 and https://github.com/brave/brave-core/pull/8062), but in this particular case I'm not sure it's a good idea.

My reasoning is that in this case using a chromium_src override to define Brave-specific values for these variables won't work as expected, since we'd need first to redefine the ones from upstream into something like `<variable_name>_ChromiumImpl` or `<variable_name>_Unused` (depending on whether we will actually use them -e.g. for the `!defined(BRAVE_CHROMIUM_BUILD)` case- or not), but doing that will also change the places where such variables are already being referenced inside this `updater_state_win.cc` file.

If this file didn't contain any logic referencing those variables and only contained definitions (as it's the case for the PRs linked above), I'd say this approach would work. But considering that's not the case and they are indeed being referenced, I'm not sure how to do this in a way that is actually better / less bad than using a .patch file.

WDYT? Do you have any suggestion?

Comment (iefremov) [reply patches/components-update_client-updater_state_win.cc.patch:25]: no suggestions, let's leave them alone :)

**Pull request merged**
